### PR TITLE
Make Map::wall_remove_*() use CoordsXY*

### PR DESCRIPTION
**Pull request by tupaschoal**
There are some suspicious code which I have commented with a TODO for you to take a look

### CODE REVIEW

File: src/openrct2/actions/FootpathPlaceAction.hpp [hunk 1]
```diff
@@ -133,10 +133,11 @@ DEFINE_GAME_ACTION(FootpathPlaceAction, GAME_COMMAND_PLACE_PATH, GameActionResul
                 auto zLow = _loc.z / 8;
                 auto zHigh = zLow + 4;
                 wall_remove_intersecting_walls(
-                    _loc.x, _loc.y, zLow, zHigh + ((_slope & TILE_ELEMENT_SURFACE_RAISED_CORNERS_MASK) ? 2 : 0),
+                    { _loc, zLow * 8, (zHigh + ((_slope & TILE_ELEMENT_SURFACE_RAISED_CORNERS_MASK) ? 2 : 0)) * 8 },
```
Comment (duncanspumpkin) [inline src/openrct2/actions/FootpathPlaceAction.hpp]: Move the *8 stuff into the sum on L133 and L134 and increase the 2 to 16.

File: src/openrct2/actions/FootpathPlaceAction.hpp [hunk 2]
```diff
@@ -411,10 +412,10 @@ DEFINE_GAME_ACTION(FootpathPlaceAction, GAME_COMMAND_PLACE_PATH, GameActionResul
...
         if (pathElement->IsSloped() && !(GetFlags() & GAME_COMMAND_FLAG_GHOST))
         {
-            int32_t direction = pathElement->GetSlopeDirection();
-            int32_t z = pathElement->base_height;
-            wall_remove_intersecting_walls(_loc.x, _loc.y, z, z + 6, direction_reverse(direction));
-            wall_remove_intersecting_walls(_loc.x, _loc.y, z, z + 6, direction);
+            auto direction = pathElement->GetSlopeDirection();
+            int32_t z = pathElement->GetBaseZ();
+            wall_remove_intersecting_walls({ _loc, z, z + 6 * 8 }, direction_reverse(direction));
+            wall_remove_intersecting_walls({ _loc, z, z + 6 * 8 }, direction);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/FootpathPlaceAction.hpp]: brackets around the `(6 * 8)`

File: src/openrct2/actions/SmallSceneryPlaceAction.hpp [hunk 3]
```diff
@@ -373,7 +373,8 @@ DEFINE_GAME_ACTION(SmallSceneryPlaceAction, GAME_COMMAND_PLACE_SCENERY, SmallSce
             footpath_remove_litter(_loc.x, _loc.y, targetHeight);
             if (!gCheatsDisableClearanceChecks && (scenery_small_entry_has_flag(sceneryEntry, SMALL_SCENERY_FLAG_NO_WALLS)))
             {
-                wall_remove_at(_loc.x, _loc.y, targetHeight, targetHeight + sceneryEntry->small_scenery.height);
+                // TODO: Seems like heights are in small coords and should be *8 ?
```
Comment (duncanspumpkin) [inline src/openrct2/actions/SmallSceneryPlaceAction.hpp]: `tile_element_height`  and `tile_element_water_height` returns big coordinates. This is fine. Remove comment.

File: src/openrct2/actions/WaterSetHeightAction.hpp [hunk 4]
```diff
@@ -116,7 +116,8 @@ DEFINE_GAME_ACTION(WaterSetHeightAction, GAME_COMMAND_SET_WATER_HEIGHT, GameActi
         int32_t surfaceHeight = tile_element_height(_coords);
         footpath_remove_litter(_coords.x, _coords.y, surfaceHeight);
         if (!gCheatsDisableClearanceChecks)
-            wall_remove_at_z(_coords.x, _coords.y, surfaceHeight);
+            // TODO: Seems to be missing the *8 on surfaceHeight?
```
Comment (duncanspumpkin) [inline src/openrct2/actions/WaterSetHeightAction.hpp]: This is correct remove comment.

File: src/openrct2/world/Wall.cpp [hunk 5]
```diff
@@ -33,27 +33,27 @@
...
  *  rct2: 0x006E588E
  */
-void wall_remove_at(int32_t x, int32_t y, int32_t z0, int32_t z1)
+void wall_remove_at(const CoordsXYRangedZ& wallPos)
 {
     TileElement* tileElement;
 
-    z0 /= 8;
-    z1 /= 8;
+    auto baseZ = wallPos.baseZ / 8;
```
Comment (duncanspumpkin) [inline src/openrct2/world/Wall.cpp]: Remove all of the *8 and /8 and just use the tileElement->GetBaseZ() and such functions.

File: src/openrct2/world/Wall.cpp [hunk 6]
```diff
@@ -63,35 +63,36 @@ void wall_remove_at(int32_t x, int32_t y, int32_t z0, int32_t z1)
...
+    tileElement = map_get_first_element_at(wallPos);
     if (tileElement == nullptr)
         return;
     do
     {
         if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
             continue;
 
-        if (tileElement->clearance_height <= z0 || tileElement->base_height >= z1)
+        // TODO: It looks like base and clearance are swapped in here
```
Comment (duncanspumpkin) [inline src/openrct2/world/Wall.cpp]: They are purposely swapped. You are looking for intersections so you compare the base of one item to see if it is below the clearance of another item.

File: src/openrct2/actions/SmallSceneryPlaceAction.hpp (hunk 3 above)
Comment (tupaschoal) [reply src/openrct2/actions/SmallSceneryPlaceAction.hpp]: Ok, thanks for explaining!

File: src/openrct2/world/Wall.cpp (hunk 6 above)
Comment (tupaschoal) [reply src/openrct2/world/Wall.cpp]: Makes sense!

**Comment by tupaschoal** [comment]
Delt with all the pending issues.

**Pull request merged**